huaweicloud/services/acceptance/resource_check.go

```diff
@@ -0,0 +1,233 @@
+package acceptance
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+// ServiceFunc the resource query functions
+type ServiceFunc func(*config.Config, *terraform.ResourceState) (interface{}, error)
+
+// ResourceCheck resource check object
+type ResourceCheck struct {
+	resourceName    string
+	resourceObject  interface{}
+	getResourceFunc ServiceFunc
+	resourceType    string
+}
+
+const (
+	resourceTypeCode   = "resource"
+	dataSourceTypeCode = "dataSource"
+
+	checkAttrRegexpStr = `^\$\{([^\}]+)\}$`
+)
+
+/*
+InitDataSourceCheck build a 'ResourceCheck' object. Only used to check datasource attributes.
+  Parameters:
+    resourceName: The resource name is used to check in the terraform.State. e.g. huaweicloud_waf_domain.domain_1
+  Return:
+    *ResourceCheck: ResourceCheck object
+*/
+func InitDataSourceCheck(sourceName string) *ResourceCheck {
```

Comment:
The comment name and the argument name are not equal. pls check other places.

Reply:
fixed